src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -336,18 +337,39 @@ private void nullifyWatchdog() {
     private String getLine(CharBuffer cb) {
```

Comment:
Maybe makes sense to check if the message fits the buffer and just call StringBuilder if it does not. Otherwise these is a kind of dark magic in the code, which seems to hammer performance with no benefit